src/perftest_parameters.h

```diff
@@ -178,8 +178,8 @@
 #define UC_MAX_RX     (16000)
 #define MIN_CQ_MOD    (1)
 #define MAX_CQ_MOD    (1024)
-#define MAX_INLINE    (912)
-#define MAX_INLINE_UD (884)
+#define MAX_INLINE    (1024)
```

Comment:
I saw your response to sshaulnv and I debugged it, When exceeding  those values the qp creation in ctx_qp_create method is failing and returning null with errno 28 and the max_inline_data in the attr is equal to the inline_size specified by the user so its not even entering the if condition that checks the inline_size.
Can you please send your setup information?

Reply:
I test it in a 5.10 kernel and rdma-core v41 version, it works well for our hns roce. 

And I tested the ConnectX-4 Lx in the same environment, and the interesting thing is that it still works well beyond the original limit. Until the test to the length of 1005, it has the same problem as you describe

The normal log:
```bash
[root@localhost perftest]# ib_send_lat -d mlx5_0 -I 1004 127.0.0.1 &
[2] 1766247
[root@localhost perftest]# ---------------------------------------------------------------------------------------
                    Send Latency Test
 Dual-port       : OFF          Device         : mlx5_0
 Number of qps   : 1            Transport type : IB
 Connection type : RC           Using SRQ      : OFF
 PCIe relax order: ON
---------------------------------------------------------------------------------------
                    Send Latency Test
 Dual-port       : OFF          Device         : mlx5_0
 Number of qps   : 1            Transport type : IB
 Connection type : RC           Using SRQ      : OFF
 PCIe relax order: ON
 ibv_wr* API     : ON
 TX depth        : 1
 Mtu             : 1024[B]
 Link type       : Ethernet
 GID index       : 3
 Max inline data : 1004[B]
 rdma_cm QPs     : OFF
 Data ex. method : Ethernet
---------------------------------------------------------------------------------------
 ibv_wr* API     : ON
 RX depth        : 512
 Mtu             : 1024[B]
 Link type       : Ethernet
 GID index       : 3
 Max inline data : 1004[B]
 rdma_cm QPs     : OFF
 Data ex. method : Ethernet
---------------------------------------------------------------------------------------
 local address: LID 0000 QPN 0x0133 PSN 0x9a60a6
 local address: LID 0000 QPN 0x0132 PSN 0x432e72
 GID: 00:00:00:00:00:00:00:00:00:00:255:255:192:168:100:100
 GID: 00:00:00:00:00:00:00:00:00:00:255:255:192:168:100:100
 remote address: LID 0000 QPN 0x0132 PSN 0x432e72
 remote address: LID 0000 QPN 0x0133 PSN 0x9a60a6
 GID: 00:00:00:00:00:00:00:00:00:00:255:255:192:168:100:100
 GID: 00:00:00:00:00:00:00:00:00:00:255:255:192:168:100:100
---------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------
 #bytes #iterations    t_min[usec]    t_max[usec]  t_typical[usec]    t_avg[usec]    t_stdev[usec]   99% percentile[usec]   99.9% percentile[usec]
 #bytes #iterations    t_min[usec]    t_max[usec]  t_typical[usec]    t_avg[usec]    t_stdev[usec]   99% percentile[usec]   99.9% percentile[usec]
 2       1000          1.08           3.19         1.20                1.20             0.06            1.38                    3.19
---------------------------------------------------------------------------------------
 2       1000          1.08           3.91         1.19                1.20             0.06            1.40                    3.91
---------------------------------------------------------------------------------------

[1]-  Done                    ib_send_lat -d mlx5_0 -I 1004
[2]+  Done                    ib_send_lat -d mlx5_0 -I 1004 127.0.0.1
```
The exception log：
```bash
[root@localhost perftest]# ib_send_lat -d mlx5_0 -I 1005 &
[1] 1766854
[root@localhost perftest]#
************************************
* Waiting for client to connect... *
************************************

[root@localhost perftest]# ib_send_lat -d mlx5_0 -I 1005 127.0.0.1 &
[2] 1766865
[root@localhost perftest]# Unable to create QP.
Failed to create QP.
 Couldn't create IB resources
Unable to create QP.
Failed to create QP.
 Couldn't create IB resources

[1]-  Exit 1                  ib_send_lat -d mlx5_0 -I 1005
[2]+  Exit 1                  ib_send_lat -d mlx5_0 -I 1005 127.0.0.1
```

I guess this problem should not be directly related to my mr. This may be caused by some limitations of mlx5. I briefly looked at the code of mlx5, and I feel that the most likely possibility is that the ENOSPC error is reported in the `alloc_var_entry()`.

Reply:
> I guess this problem should not be directly related to my pr. This may be caused by some limitations of mlx5. I briefly looked at the code of mlx5, and I feel that the most likely possibility is that the ENOSPC error is reported in the `alloc_var_entry()`.

The `alloc_var_entry()` seems to be for DV(Direct verbs) and I'm not familiar with that. So,  I use CM to test, so that the mlx5 driver enters the traditional create qp path. Like this:
```bash
ib_send_lat -d mlx5_0 -I 1024 -R &
ib_send_lat -d mlx5_0 -I 1024 192.168.100.100 -R &
```

Then I analyzed the problem and finally found that the problem in the CM scenario can be solved by the following patch (for rdma-core).
```patch
diff --git a/providers/mlx5/verbs.c b/providers/mlx5/verbs.c
index 0331ea9..0194d44 100644
--- a/providers/mlx5/verbs.c
+++ b/providers/mlx5/verbs.c
@@ -1597,6 +1597,8 @@ static int mlx5_calc_send_wqe(struct mlx5_context *ctx,
        if (attr->cap.max_inline_data) {
                inl_size = size + align(sizeof(struct mlx5_wqe_inl_data_seg) +
                        attr->cap.max_inline_data, 16);
+               if (inl_size > ctx->max_sq_desc_sz)
+                       inl_size = ctx->max_sq_desc_sz;
        }

        if (attr->comp_mask & IBV_QP_INIT_ATTR_MAX_TSO_HEADER) {
```

Reply:
Hi，@HassanKhadour，From the above phenomenon, it seems that the problem is not directly related to the change of max inline size. Is this patch acceptable?